app/domain/authentication/common/extract_resource_restrictions.rb

```diff
@@ -0,0 +1,57 @@
+require 'command_class'
+
+module Authentication
+  module Common
+
+    ExtractResourceRestrictions = CommandClass.new(
+        dependencies: {
+            role_class:     ::Role,
+            resource_class: ::Resource,
+            logger:         Rails.logger
+        },
+        inputs:   %i(authenticator_name service_id host_name account)
+    ) do
+
+      def call
+        fetch_all_role_annotations
+        convert_to_resource_restrictions
+        @resource_restrictions
+      end
+
+      private
+
+      def fetch_all_role_annotations
```

Comment:
i would call this `fetch_role_annotations`. it is clear that all the annotations will be fetched.